Source/WebCore/dom/Subscriber.h

```diff
@@ -50,7 +50,9 @@ class Subscriber final : public ActiveDOMObject, public ScriptWrappable, public
     void addTeardown(Ref<VoidCallback>);
 
     bool active() { return m_active; }
-    AbortSignal& signal() { return m_abortController->signal(); }
+    AbortSignal& signal() { return m_signal.get(); }
```

Comment:
This is referenced by the IDL. So we need a method called `signal`. But question, (cc @darinadler @cdumez @rreno)

Is it possible through the IDL bindings, to map the `signal` IDL to an implemented method of `protectedSignal` — and only have the one here, that is protected.

Or would this be totally fine to do:

```cpp
// For the IDL binding
Ref<AbortSignal> signal() { return protectedSignal(); }
// ^^^^^^^^^^^^^ can we bind Ref to the idl?

// The real method we should be using internally
Ref<AbortSignal> protectedSignal() const { return m_signal; }
```

Reply:
> Ref<AbortSignal> signal() const { return m_signal; }

would not be fine as it doesn't follow our pattern. We usually have both `signal()` which returns a raw pointer/reference and `protectedSignal()` which returns a Ref/RefPtr. It is up to the call site to decide which one to call based on their use case. If the call site only need to call a trivial getter on the signal, then it doesn't need to call `protectedSignal()`, `signal()` will suffice. I do not believe the generated JS bindings NEED a smart pointer here for safety (or the cpp bot would complain).